apps/mymedicare_cb/tests/test_callback_slsx.py

```diff
@@ -434,3 +429,467 @@ def catchall(url, request):
+
+        Two related/opposite tests were removed from test_callback_exceptions() and relocted here.
+
+        TEST SUMMARY:
+
+            NOTE: The fhir_patient_info_mock always provides a BFD patient resource search match to
+                  a bene with fhir_id = -20140000008325
+
+
+            1. First successful matching for beneficiary having valid only hicn and EMPTY mbi.
+               This creates a new Crosswalk entry with hicn and NULL mbi hash values used in the initial match.
+
+            2. The bene's MBI has been changed from empty to valid value in the mock SLSx user_info response.
+
+            3. Remove Crosswalk and ArchivedCrosswalk entries to start fresh.
+
+            4. First successful matching for beneficiary having valid hicn/mbi.
+               This creates a new Crosswalk entry with hicn/mbi hash values used in the initial match.
+
+            5. The bene's HICN has been changed in the mock SLSx user_info response.
+
+               This is for the use case where a beneficary's HICN has been changed in the
+               SLSx/BEDAP upstream identity service.
+
+               This response is mocked by:  MockUrlSLSxResponses.slsx_user_info_mock_changed_hicn
+
+               This would previously FAIL with response.status_code == status.HTTP_500_INTERNAL_SERVER_ERROR
+               with error text: "Found user's hicn did not match".
+
+            6. Restore saved_hicn_hash in Crosswalk prior to next test. Restore crosswalk state to same as #3.
+
+            7. The bene's MBI has been changed in the mock SLSx user_info response.
+               This response is mocked by:  MockUrlSLSxResponses.slsx_user_info_mock_changed_mbi
+
+            8. Restore saved_mbi_hash in Crosswalk prior to next test. Restore crosswalk state to same as #3.
+
+            9. The bene's HICN & MBI (both) have been changed in the mock SLSx user_info response.
+               This response is mocked by:  MockUrlSLSxResponses.slsx_user_info_mock_changed_hicn_mbi
+        '''
+        # create a state
+        state = generate_nonce()
+        AnonUserState.objects.create(
+            state=state,
+            next_uri="http://www.google.com?client_id=test&redirect_uri=test.com&response_type=token&state=test")
+
+        # mock fhir patient endpoint (back end bfd) with fhir_id == "-20140000008325"
+        @urlmatch(netloc="fhir.backend.bluebutton.hhsdevcloud.us", path="/v1/fhir/Patient/")
+        def fhir_patient_info_mock(url, request):
+            return {
+                "status_code": status.HTTP_200_OK,
+                "content": patient_response,
+            }
+
+        @all_requests
+        def catchall(url, request):
+            raise Exception(url)
+
+        # 1. First successful matching for beneficiary having valid only hicn and EMPTY mbi.
+        with HTTMock(MockUrlSLSxResponses.slsx_token_mock,
+                     MockUrlSLSxResponses.slsx_user_info_empty_mbi_mock,
```

Comment:
Note `MockUrlSLSxResponses.slsx_user_info_empty_mbi_mock` difference. This will change in later tests as well, depending on what type of mock response is needed.